### PR TITLE
Add metadata recursive endpoint

### LINKED ISSUE #2603: Feature Request: recursive endpoint to read metadata
It would be really handy if an inscription could hit an endpoint to read its own metadata.

### PR DESCRIPTION
**Pull request by rot13maxi**
closes #2603

quick little endpoint. open to change the path, or whatever other feedback folks have :)

### CODE REVIEW

**Comment by elocremarc** [comment]
Oh nice!
@raphjaph lets just yeet into /-/ and get the endpoints moving.

@rot13maxi can you test out if this works on regtest via recursion by adding the CSP-Headers like Raphs pr? Wonder if we should all rebase off his?

File: src/subcommand/server.rs [hunk 1]
```diff
@@ -694,6 +695,51 @@ impl Server {
     )
   }
 
+  async fn metadata(
+    Extension(index): Extension<Arc<Index>>,
+    Extension(config): Extension<Arc<Config>>,
+    Path(inscription_id): Path<InscriptionId>,
+  ) -> ServerResult<Response> {
```
Comment (elocremarc) [inline src/subcommand/server.rs:702]: Should return json
```diff
-  ) -> ServerResult<Response> {
+ ) -> ServerResult<Json<String>> {
```

Comment (elocremarc) [inline src/subcommand/server.rs:702]: Response should be type Json?

**Comment by casey** [comment]
I think for metadata, we should return the CBOR. Converting the CBOR to JSON makes the details of that conversion part of the contract of the endpoint. The mapping from CBOR to JSON isn't well defined, and I'm not sure how serde would handle types in the CBOR which have no JSON equivalent. It sucks to make inscriptions deserialize CBOR, but someone can always inscribe a CBOR parser, and everyone can use that, and when something goes wrong, we can blame the library author instead of us 😂

For the response, I would just return a string with the hex-encoded CBOR, since all recursive endpoint responses must be JSON.

**Comment by rot13maxi** [comment]
Hilariously I originally just returned cbor, but then i saw raph was trying to jsonify all the things so changed it. Im happy to change it back and make deserialzation someone else’s problem.

**Comment by devords** [comment]
Instead of metadata specific endpoint, how about a recursive endpoint for inscription that includes metadata among other things like current owner, sat location etc? Basically, whatever we can include without needing a full sat index.

**Review by raphjaph — APPROVED** [review]
LGTM!

**Pull request merged**